lib/watcher.js

```diff
@@ -244,12 +244,17 @@ class Watcher {
 		}
 	}
 
-	trackFailures(api) {
+	trackFailures(api, reporter) {
 		api.on('run', plan => {
 			this.pruneFailures(plan.files);
 
 			const currentVector = this.runVector;
 			plan.status.on('stateChange', evt => {
+				if (evt.type === 'interrupt') {
+					reporter.endRun();
+					process.exit(1); // eslint-disable-line unicorn/no-process-exit
```

Comment:
Why does this explicitly exit (with an exit code), but when AVA runs without the watcher it doesn't?

Reply:
Good catch. I'll make it exit(1) if ran without the watcher as well.